lib/matcher.js

```diff
@@ -1,3 +1,4 @@
+var _ = require('lodash');
 var PROPERTY_TO_FLAG_AS_STRUMMER_MATCHER = "IS-STRUMMER-MATCHER";
 
 function Matcher(opts) {
```

Comment:
Good point @nguyenchr, I don't think it handles objects that happen to have a property called optional :cry: 

```js
var attribute = s.object({
  name: 'string',
  value: 'number',
  optional: 'bool'
})
```